### PR TITLE
Add leadership council

### PR DESCRIPTION
**Pull request by jackh726**
Officially establishes the Leadership Council. Also, archives interim-leadership-chat and removes core team from website.

### CODE REVIEW

**Comment by ehuss** [comment]
Would it make sense to also set up a Zulip group with `[[zulip-groups]]`?

File: teams/leadership-council.toml
Comment (rylev) [inline teams/leadership-council.toml:40]: What is the purpose of this? If we decide this is necessary we should quickly document what this is used for (and eventually document in more detail somewhere else all the ways one can get in touch with the council).

**Review by rylev — CHANGES_REQUESTED** [review]
Looks good but waiting on the RFC to be merged *and* for Zulip user group configuration to be added before merging this.

File: teams/leadership-council.toml
Comment (pietroalbini) [inline teams/leadership-council.toml:20]: The "team" bors review permissions is not really used (this repo doesn't use bors).

**Review by rylev — APPROVED** [review]
Going to merge this once CI passes. I can add the Zulip user group in a follow up PR.

**Comment by jackh726** [comment]
> I can add the Zulip user group in a follow up PR.

I did add the user group!

**Comment by rylev** [comment]
@jackh726 can you rebase against master now?

File: teams/leadership-council.toml
Comment (rylev) [inline teams/leadership-council.toml:36]: Nit: should we use `T-council`? The council isn't necessarily a team in the traditional sense though...

Comment (jackh726) [reply teams/leadership-council.toml:36]: The *stream* is just `council`, so having the *group* just be `council` keeps it consistent.

**Comment by cbeuw** [comment]
You might want to add the localization strings in https://github.com/rust-lang/www.rust-lang.org/tree/master/locales!

```html
<title>
    Unknown localization governance-team-leadership-council-name - Rust Programming Language
</title>
```

**Pull request merged**
